WordPressLoginFlow/src/main/java/org/wordpress/android/login/Login2FaFragment.java

```diff
@@ -537,4 +593,65 @@ private void setTextForSms() {
         mOtpButton.setText(getString(R.string.login_text_otp_another));
         mSentSmsCode = true;
     }
+
+    private void doAuthWithSecurityKeyAction() {
+        mAnalyticsListener.trackUseSecurityKeyClicked();
+        if (!NetworkUtils.checkConnection(getActivity())) {
+            return;
+        }
+
+        startProgress();
+        mOldSitesIDs = SiteUtils.getCurrentSiteIds(mSiteStore, false);
+
+        StartWebauthnChallengePayload payload = new StartWebauthnChallengePayload(
+                mUserId, mWebauthnNonce);
+        mDispatcher.dispatch(AuthenticationActionBuilder
+                .newStartSecurityKeyChallengeAction(payload));
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onWebauthnChallengeReceived(WebauthnChallengeReceived event) {
+        if (event.isError()) {
+            endProgress();
+            handleAuthError(event.error.type, getString(R.string.login_error_security_key));
+            return;
+        }
+        mPasskeyCredentialsHandler = new PasskeyCredentialsHandler(
+                event.mUserId,
+                event.mChallengeInfo
+        );
+        mPasskeyCredentialsHandler.createIntentSender(
+                requireContext(),
+                intent -> {
+                    assert mResultLauncher != null;
```

Comment:
@ThomazFB - using `assert` feels off; would do you think about using an `if` instead? Same for the two `assert` statements below. iirc, assertions may be disabled at runtime in Java.

Reply:
Good point about the asserts! I also took advantage of the ifs and placed an error handling in case something go wrong. Fixed in https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/123/commits/a681add8039999e23fcbd36d31b14fe3d5c99980.